app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -2344,11 +2350,29 @@ class BrowserTabFragment :
                     }
                 },
             )
+            addNoOpWebMessageListener(it)
         }
 
         WebView.setWebContentsDebuggingEnabled(webContentDebugging.isEnabled())
     }
 
+    // See https://app.asana.com/0/1200204095367872/1207300292572452/f (WebMessageListener debugging)
+    private fun addNoOpWebMessageListener(webView: DuckDuckGoWebView) {
+        if (WebViewFeature.isFeatureSupported(WebViewFeature.WEB_MESSAGE_LISTENER) &&
```

Comment:
It's possible that `webMessageListenerFeature.self().isEnabled()` will read from disk (if first time calling it) and can contribute to ANRs since this is being called on the main thread.

I'm thinking that doing this check off of the main thread is good generally here, but also good because it's a better simulation of what we'll likely have to do for at least some _real_ web message listeners where we have to do some checks on an `io` thread first before deciding to add the listeners.

Reply:
Everything in this function is now running on `io`: https://github.com/duckduckgo/Android/pull/4551/commits/9f6a4a7948d3131af26977cdbebe69b1e680600c

Reply:
I’ve moved the feature check to `io` and the rest to `main`: https://github.com/duckduckgo/Android/pull/4551/commits/db9bbb997afec5dfb03cbf47e060126214f8351e